EntAppProject/src/main/java/com/jobook/EntAppProject/Services/JobServiceStub.java

```diff
@@ -9,49 +9,56 @@
 import com.jobook.EntAppProject.model.Customer;
 import com.jobook.EntAppProject.model.Employee;
 import com.jobook.EntAppProject.model.Job;
+import org.springframework.util.Assert;
 
 @Component
 public class JobServiceStub implements IJobService {
-	
+
+	private final IUserService userService;

```

Comment:
Only use field injection when necessary. The reason for this being that field injection makes your classes kind of worthless without dependency injection. Also, you let yourself/others instantiate the class without its dependencies, which is a NullReferenceException waiting to happen. Use constructor injection whenever you can

Reply:
I could go either way on this.  I tend to use field injection... it's usually injected when needed, and de-clutters the constructor.  The trick is, if your constructor is full of parameters, and then you subclass this class, you have to deal with a nasty super() call to satisfy those parameters.

Reply:
I just do not like field injection unless it is absolutely needed, because then you are allowing the creation of a class without the services it needs, and it is not as clear that the class requires dependencies. I guess you could argue the same with constructor injection, though.